### PR TITLE
Pure ftp drop

### PR DESCRIPTION
**Pull request by jreidinger**
https://trello.com/c/uL3kzh2z/1322-8-remove-yast-xinetd-autoyast-ftp-server-and-inetd-module

after approval another round of manual testing is planned. The first round was successful on SLE15.

### CODE REVIEW

File: package/yast2-ftp-server.changes [hunk 1]
```diff
@@ -1,3 +1,12 @@
+-------------------------------------------------------------------
+Thu Feb  8 15:07:37 UTC 2018 - jreidinger@suse.com
+
+- convert starting on demand from Xinetd to systemd sockets due
+  to drop of Xinetd (fate#323373)
+- Drop pure-ftp as it does not provide systemd socket and we want
+  to support only one backend to reduce maintenance loadout
```
Comment (teclator) [inline package/yast2-ftp-server.changes]: `maintenance loadout` => `the maintenance charge`

File: package/yast2-ftp-server.spec [hunk 2]
```diff
@@ -17,15 +17,14 @@
...
-Version:        4.0.0
+Version:        4.0.1
 Release:        0
 
 BuildRoot:      %{_tmppath}/%{name}-%{version}-build
 Source0:        %{name}-%{version}.tar.bz2
 
 # SuSEFirewall2 replace by firewalld (fate#323460)
 Requires:       yast2 >= 4.0.39
-Requires:       yast2-inetd
```
Comment (teclator) [inline package/yast2-ftp-server.spec]: As we only support one backend `pure-ftpd` can be removed from the description.

File: src/clients/ftp-server.rb [hunk 3]
```diff
@@ -355,25 +313,25 @@ def main
               "Anonymous users can create directories."
             )
           },
-          #only vsftd
+          # only vsftd
```
Comment (teclator) [inline src/clients/ftp-server.rb]: NP: We were removing allusions to `only vsftpd` below, so why not just remove them completely

File: src/clients/ftp-server.rb [hunk 4]
```diff
@@ -689,18 +597,15 @@ def FTPdCMDStartup(options)
...
+      elsif !Ops.get(options, "socket").nil?
         CommandLine.Print("")
         # TRANSLATORS: CommandLine header
         CommandLine.Print(String.UnderlinedHeader(_("Start-Up:"), 0))
         CommandLine.Print("")
         # TRANSLATORS: CommandLine progress information
-        CommandLine.Print(_("Start FTP daemon via xinetd"))
+        CommandLine.Print(_("Start FTP daemon via socket"))
         CommandLine.Print("")
         SetStartedViaXinetd(true)
```
Comment (teclator) [inline src/clients/ftp-server.rb]: Shouldn't this call be removed?

File: src/clients/ftp-server.rb [hunk 5]
```diff
@@ -443,20 +387,19 @@ def main
...
-    def FTPdCMDShow(options)
-      options = deep_copy(options)
+    def FTPdCMDShow(_options)
       CommandLine.Print("")
       CommandLine.Print(String.UnderlinedHeader(_("Display Settings:"), 0))
       CommandLine.Print("")
-      #start-up settings
+      # start-up settings
       CommandLine.PrintNoCR(_("Start-Up:"))
       if GetStartedViaXinetd()
```
Comment (teclator) [inline src/clients/ftp-server.rb]: Shouldn't this call be removed as commented below with Set method?

File: src/modules/FtpServer.rb [hunk 6]
```diff
@@ -2,19 +2,10 @@
 
 require "yast"
 
+require "y2firewall/firewalld"
```
Comment (teclator) [inline src/modules/FtpServer.rb]: Thnx!

File: src/clients/ftp-server.rb (hunk 4 above)
Comment (jreidinger) [reply src/clients/ftp-server.rb]: why it should be removed? It is CLI that inform you that FTP Daemon is starting by socket listening.

File: src/clients/ftp-server.rb (hunk 3 above)
Comment (jreidinger) [reply src/clients/ftp-server.rb]: makes sense

File: src/clients/ftp-server.rb (hunk 4 above)
Comment (teclator) [reply src/clients/ftp-server.rb]: I mean the `SetStartedViaXinetd(true)`

File: src/clients/ftp-server.rb (hunk 5 above)
Comment (jreidinger) [reply src/clients/ftp-server.rb]: why it should be removed? When systemd socket start ftp daemon, then it should be written. There are three states: socket, service and manual and this is one of the states.

File: src/clients/ftp-server.rb (hunk 4 above)
Comment (jreidinger) [reply src/clients/ftp-server.rb]: yep, make sense, now it do socket enablement.

File: src/clients/ftp-server.rb (hunk 5 above)
Comment (teclator) [reply src/clients/ftp-server.rb]: As with Set I will rename it to correspond the behavior with the naming

**Pull request merged**